src/containers/my-resources/add-attachment-category-modal/AddAttachmentCategoryModal.tsx

```diff
@@ -0,0 +1,107 @@
+import { FC, SyntheticEvent, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import Box from '@mui/material/Box'
+import Typography from '@mui/material/Typography'
+
+import AppButton from '~/components/app-button/AppButton'
+import AsyncAutocomplete from '~/components/async-autocomlete/AsyncAutocomplete'
+
+import { ResourceService } from '~/services/resource-service'
+import useForm from '~/hooks/use-form'
+
+import { getInitialValues } from '~/containers/my-resources/add-attachment-category-modal/AddAttachmentCategoryModal.constants'
+import { styles } from '~/containers/my-resources/add-attachment-category-modal/AddAttachmentCategoryModal.styles'
+import {
+  Attachment,
+  CategoryNameInterface,
+  UpdateAttachmentParams,
+  EditAttachmentForm,
+  ButtonTypeEnum,
+  ButtonVariantEnum,
+  ComponentEnum
+} from '~/types'
+
+interface AddAttachmentCategoryModalProps {
+  closeModal: () => void
+  attachment: Attachment
+  updateAttachmentCategory: (
+    params?: UpdateAttachmentParams | undefined
+  ) => Promise<void>
+}
+
+const AddAttachmentCategoryModal: FC<AddAttachmentCategoryModalProps> = ({
+  closeModal,
+  attachment,
+  updateAttachmentCategory
+}) => {
+  const { t } = useTranslation()
+
+  const [loading, setLoading] = useState<boolean>(false)
+
+  const { data, errors, handleNonInputValueChange, handleBlur, handleSubmit } =
+    useForm<EditAttachmentForm>({
+      initialValues: getInitialValues(attachment),
+      onSubmit: async () => {
+        setLoading(true)
+        await updateAttachmentCategory({
+          id: attachment._id,
+          fileName: data.fileName,
+          category: data.category
```

Comment:
I think, initialValues can only contain category, and you can pass data to category: 
await updateAttachmentCategory(data)

Reply:
I have used both fileName and description because the service for attachment update on backend requires them as parameters, so I have to send this data in a request. 
I have changed updateAttachmentCategory and pass only data object. 

<img width="573" alt="Знімок екрана 2023-11-23 о 19 42 46" src="https://github.com/ita-social-projects/SpaceToStudy-Client/assets/17857767/a300122c-e799-4a6e-9112-47ecc9da32fc">

Reply:
it shouldn`t be required, I think you can add check for description in attachment service on backend and pass only category: 

if(description ) {
attachment.description = description
}

Reply:
> They shouldn`t be required, I think you can add check for description in attachment service on backend and pass only category:
> 
> if(description ) { attachment.description = description }

done)